### ISSUE TITLE
Currently the demo site doesn't work because RTCRtpTransceiver is not defined in Edge

### ISSUE THREAD

**Issue by soundlake** [meetecho/janus-gateway#1759]
Janus version: not sure, but it is used in the demo page.
Edge version: Microsoft Edge 44.18970.1001.0, Microsoft EdgeHTML 18.18970
Error message: SCRIPT5009: SCRIPT5009: 'RTCRtpTransceiver' is not defined (at 
janus.js (393,10))

I already searched with "Edge", and I'm a bit surprised once Edge was supported.

**Comment by lminiero** [comment]
Is this the "old" Edge, or the new Chromium-based one? I don't use Windows so can't test myself.

**Comment by lminiero** [comment]
Can you check if changing that line to this fixes it for you?

		} else if(!RTCRtpTransceiver || !('currentDirection' in RTCRtpTransceiver.prototype)) {

My guess is you're getting the error because we try to access the `prototype` property of something that does not exist.

**Comment by lminiero** [comment]
Pushed the fix already, so you can test master. Feel free to reopen if still an issue.

**Comment by soundlake** [comment]
Thank you for the quick fix. However, unfortunately, it doesn't work. 🙁 

I'm not sure since when EdgeHTML changed this, but casting `undefined` to a `boolean` with `!` doesn't just work. 😡 

```js
> !undefined  // true
> x  // undefined
> !x  // 'x' is not defined
> Boolean(x)  // 'x' is not defined
```

Thus, to work around this Edge's peculiarity, we need
```js
} else if {typeof RTCRtpTransceiver !== 'undefined' || !('currentDirection' in RTCRtpTransceiver.prototype)) {
```

However, on the other hand, I'm not sure this issue is really something to take care of. Because the Chromium based Edge will replace this old Edge anyway.

**Comment by soundlake** [comment]
🤔 Maybe I don't have the right to reopen this issue.

According to [this StackOverflow answer](https://stackoverflow.com/a/21333938/4276533), I don't have it, because you, the repo collaborator closed this. 😉

**Comment by lminiero** [comment]
Don't you mean `typeof RTCRtpTransceiver === 'undefined'` ? If that gets Edge working, no problem for  me to add the improved check, even if it will disappear soon.

**Comment by soundlake** [comment]
@lminiero Thank you for pointing out the typo! I fixed it. And another thank you for the old Edge support 👍

**Comment by lminiero** [comment]
Can you check if the above check works now?

**Comment by soundlake** [comment]
@lminiero Thank you for your effort, but I'm sorry to tell you that it doesn't work. 🙁 

EdgeHTML seems insane to me. There seems to be 2 types of `undefined`

```js
> x  // undefined
> x === undefined  // 'x' is not defined
> x = undefined
> x === undefined  // true
```

I don't think there is another way to achieve this with `typeof RTCRtpTransceiver === 'undefined'. 🙁 

btw, fyi, `null` works as expected.
```js
> x = null
> x === null  // true
```

**Comment by lminiero** [comment]
Then please tell me exactly what makes it happy, posting the correct line or preparing a pull request. Since I can't test I can't check myself.

**Comment by soundlake** [comment]
TL; DR;

The following works. Note that the code attempts to access the `RTCRtpTransceiver` property of the `window` object.

```js
} else if(!window.RTCRtpTransceiver || !('currentDirection' in RTCRtpTransceiver.prototype)) {
```

Apparently, the error message that I got was a `ReferenceError`.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Not_defined

Check the following example:

```js
> x  // 'x' is not defined (ReferenceError)
> window.x  // undefined
> !window.x  // true
```

**Comment by lminiero** [comment]
Done, thanks :+1:

**Comment by soundlake** [comment]
Thank YOU, @lminiero!
